hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/Text.java

```diff
@@ -73,6 +73,10 @@ protected CharsetDecoder initialValue() {
     }
   };
 
+  // max size of the byte array, seems to be a safe choice for multiple VMs
+  // (see ArrayList.MAX_ARRAY_SIZE)
```

Comment:
Where is this ArrayList.MAX_ARRAY_SIZE?

Reply:
Open the implementation of java.util.ArrayList, you'll find it. Also AbstractCollection, HashTable, etc. They all use this value.